course/src/Vector3.cc

```diff
@@ -0,0 +1,117 @@
+#include "Vector3.h"
+
+namespace ekumen
+{
+
+double Vector3::norm() const
+{
+    return sqrt((x_ * x_) + (y_ * y_) + (z_ * z_));
+}
+
+double Vector3::dot(const Vector3 &q) const
+{
+    return ((x_ * q.x()) + (y_ * q.y()) + (z_ * q.z()));
+}
+
+Vector3 Vector3::cross(const Vector3 &q) const
+{
+    //Vector3 r;
+    //r[0] = y_*q.z() - z_*q.y();
+    //r[1] = z_*q.x() - x_*q.z();
+    //r[2] = x_*q.y() - y_*q.x();
+    //not implemented
+    return (Vector3(y_ * q.z() - z_ * q.y(), z_ * q.x() - x_ * q.z(), x_ * q.y() - y_ * q.x())); //only for shutdown warning
+}
+
+Vector3 operator+(const Vector3 &p, const Vector3 &q)
+{
+    Vector3 res(p.x() + q.x(), p.y() + q.y(), p.z() + q.z());
+    return res;
+}
+
+Vector3 operator-(const Vector3 &p, const Vector3 &q)
+{
+    Vector3 res(p.x() - q.x(), p.y() - q.y(), p.z() - q.z());
+    return res;
+}
+
+Vector3 operator*(const double &cte, const Vector3 &p)
+{
+    Vector3 res(cte * p.x(), cte * p.y(), cte * p.z());
+    return res;
+}
+
+Vector3 operator*(const Vector3 &p, const double &cte)
+{
+    Vector3 res(cte * p.x(), cte * p.y(), cte * p.z());
+    return res;
+}
+
+Vector3 operator*(const Vector3 &p, const Vector3 &q)
+{
+    Vector3 res(p.x() * q.x(), p.y() * q.y(), p.z() * q.z());
+    return res;
+}
+
+Vector3 operator/(const Vector3 &p, const Vector3 &q)
+{
+    Vector3 res(p.x() / q.x(), p.y() / q.y(), p.z() / q.z());
+    return res;
+}
+
+bool operator==(const Vector3 &p, const Vector3 &q)
+{
```

Comment:
Use the clang-format formatter, and if you're using vscode, you can add a plugin to use it from the interface to autoformat everything.